pkg/apis/batch/validation/validation.go

```diff
@@ -389,9 +401,130 @@ func validatePodFailurePolicyRuleOnExitCodes(onExitCode *batch.PodFailurePolicyO
 	return allErrs
 }
 
+func validateSuccessPolicy(spec *batch.JobSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	criteriaPath := fldPath.Child("criteria")
+	if len(spec.SuccessPolicy.Criteria) == 0 {
+		allErrs = append(allErrs, field.Required(criteriaPath, "at least one criteria must be specified when the successPolicy is specified"))
+	}
+	for i, criterion := range spec.SuccessPolicy.Criteria {
+		allErrs = append(allErrs, validateSuccessPolicyCriteria(spec, &criterion, criteriaPath.Index(i))...)
+	}
+	return allErrs
+}
+
+func validateSuccessPolicyCriteria(spec *batch.JobSpec, criteria *batch.SuccessPolicyCriteria, criteriaPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if criteria.SucceededCount == nil && criteria.SucceededIndexes == nil {
+		allErrs = append(allErrs, field.Required(criteriaPath, "at least one of succeededCount or succeededIndexes must be specified"))
+	}
+	var totalIndexes *int
+	if criteria.SucceededIndexes != nil {
+		succeededIndexes := criteriaPath.Child("succeededIndexes")
+		if len(*criteria.SucceededIndexes) > maxJobSuccessPolicySucceededIndexesLimit {
+			allErrs = append(allErrs, field.TooLong(succeededIndexes, *criteria.SucceededIndexes, maxJobSuccessPolicySucceededIndexesLimit))
+		}
+		var err error
+		if totalIndexes, err = validateIndexesFormat(*criteria.SucceededIndexes, int(*spec.Completions)); err != nil {
+			allErrs = append(allErrs, field.Invalid(succeededIndexes, *criteria.SucceededIndexes, fmt.Sprintf("error parsing succeededIndexes: %s", err.Error())))
+		}
+	}
+	if criteria.SucceededCount != nil {
+		succeededCountPath := criteriaPath.Child("succeededCount")
+		if *criteria.SucceededCount < 0 {
+			allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*criteria.SucceededCount), succeededCountPath)...)
+		}
+		if *criteria.SucceededCount > *spec.Completions {
+			allErrs = append(allErrs, field.Invalid(succeededCountPath, *criteria.SucceededCount, fmt.Sprintf("must be less than or equal to %d (the number of specified completions)", *spec.Completions)))
+		}
+		if totalIndexes != nil && int(*criteria.SucceededCount) > *totalIndexes {
+			allErrs = append(allErrs, field.Invalid(succeededCountPath, *criteria.SucceededCount, fmt.Sprintf("must be less than or equal to %d (the number of indexes in the specified succeededIndexes field)", *totalIndexes)))
+		}
+	}
+	return allErrs
+}
+
+func IsJobFinished(job *batch.Job) bool {
```

Comment:
expected this to drop out of the diff. missing rebase?

Reply:
Yea, I think so too. I'll try to rebase again.

Reply:
This was resolved.